component/src/main/java/org/wso2/extension/siddhi/map/avro/sinkmapper/AvroSinkMapper.java

```diff
@@ -0,0 +1,217 @@
+                                "The schema should be specified as a quoted json string.",
+                        type = {DataType.STRING})
+        },
+        examples = {
+                @Example(
+                        syntax = "@sink(type='inMemory', topic='stock', @map(type='avro'," +
+                                 "schema.def = \"\"\"{\"type\":\"record\",\"name\":\"stock\"," +
+                                "\"namespace\":\"stock.example\",\"fields\":[{\"name\":\"symbol\"," +
+                                "\"type\":\"string\"},{\"name\":\"price\":\"type\":\"float\"}," +
+                                 "{\"name\":\"volume\",\"type\":\"long\"}]}\"\"\"))\n" +
+                                 "define stream stockStream (symbol string, price float, volume long);",
+                        description = "The above configuration performs a default Avro mapping that "
+                                + "generates an Avro message as output byte array.")
+        }
+)
+
+public class AvroSinkMapper extends SinkMapper {
+    private static final Logger log = Logger.getLogger(AvroSinkMapper.class);
+    private static final String DEFAULT_AVRO_MAPPING_PREFIX = "schema";
+    private static final String SCHEMA_IDENTIFIER = "def";
+    private static final String UNDEFINED = "undefined";
+
+    private String[] attributeNameArray;
+    private Schema schema;
+
+    /**
+     * Initialize the mapper and the mapping configurations.
+     *
+     * @param streamDefinition          The stream definition
+     * @param optionHolder              Option holder containing static and dynamic options
+     * @param payloadTemplateBuilderMap Unmapped list of payloads for reference
+     */
+    @Override
+    public void init(StreamDefinition streamDefinition, OptionHolder optionHolder, Map<String, TemplateBuilder>
+            payloadTemplateBuilderMap, ConfigReader mapperConfigReader, SiddhiAppContext siddhiAppContext) {
+
+        this.attributeNameArray = streamDefinition.getAttributeNameArray();
+        //if @payload() is added there must be at least 1 element in it, otherwise a SiddhiParserException raised
+        if (payloadTemplateBuilderMap != null && payloadTemplateBuilderMap.size() != 1) {
+            throw new SiddhiAppCreationException("Avro sink-mapper does not support multiple @payload mappings, " +
+                    "error at the mapper of '" + streamDefinition.getId() + "'");
+        }
+        if (payloadTemplateBuilderMap != null &&
+                payloadTemplateBuilderMap.get(payloadTemplateBuilderMap.keySet().iterator().next()).isObjectMessage()) {
+            throw new SiddhiAppCreationException("Avro sink-mapper does not support object @payload mappings, " +
+                    "error at the mapper of '" + streamDefinition.getId() + "'");
+        }
+        schema = getAvroSchema (optionHolder.validateAndGetStaticValue(DEFAULT_AVRO_MAPPING_PREFIX.concat(".").
```

Comment:
Option does not throw an exception, because a default value of null is returned in case the mapping is not there. This is because, having if this mapping is not there, the alternative methods of schema generation should be used.
I will change the getAvroSchema method to throw an exception, without returning null.

Reply:
Changes done in commit 0f7e226066bf0c99b15420226b907d1b4d5c5539